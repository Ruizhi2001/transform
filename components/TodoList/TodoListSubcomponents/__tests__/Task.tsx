import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import Task from "../Task";

describe("Task component", () => {
  it("should render with the provided text", () => {
    const text = "Sample task";
    const { getByText } = render(<Task text={text} />);

    const taskText = getByText(text);
    expect(taskText).toBeTruthy();
  });

  it("should render a colored square", () => {
    const { getByTestId } = render(<Task text="Sample task" />);

    const square = getByTestId("colored-square");
    expect(square).toBeTruthy();
  });

  it("should render a circular element", () => {
    const { getByTestId } = render(<Task text="Sample task" />);

    const circular = getByTestId("circular-element");
    expect(circular).toBeTruthy();
  });

  it("matches the snapshot", () => {
    const tree = renderer.create(<Task text="Sample task" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
