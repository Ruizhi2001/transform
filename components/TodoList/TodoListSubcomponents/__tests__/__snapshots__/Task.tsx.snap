// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderRadius": 10,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 20,
      "padding": 15,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#55BCF6",
          "borderRadius": 5,
          "height": 24,
          "marginRight": 15,
          "opacity": 0.4,
          "width": 24,
        }
      }
      testID="colored-square"
    />
    <Text
      style={
        {
          "maxWidth": "80%",
        }
      }
    >
      Sample task
    </Text>
  </View>
  <View
    style={
      {
        "borderColor": "#55BCF6",
        "borderRadius": 5,
        "borderWidth": 2,
        "height": 12,
        "width": 12,
      }
    }
    testID="circular-element"
  />
</View>
`;
